src/utils/CircleCI.ts

```diff
@@ -57,16 +62,49 @@ export class CircleCI {
    * @param orgID - The organization ID
    * @returns An array of repos
    */
-  async getRepos(orgID: string): Promise<CircleCIRepo[]> {
-    const repos = await this._getPaginated<CircleCIRepo>(
+  async getRepos(orgID: string): Promise<CircleCIAPIRepo[]> {
+    const repos = await this._getPaginated<CircleCIAPIRepo>(
       `${CircleCI.endpoint.private}/project?organization-id=${orgID}`
     );
     return repos;
   }
+
+  async getContexts(orgID: string, slug: string): Promise<CircleCIContext[]> {
+    printMessage("...", "Fetching contexts");
+    const contextsReport: CircleCIContext[] = [];
+    const contexts = await this._getPaginated<CircleCIAPIContext>(
+      `${CircleCI.endpoint.v2}/context?owner-id=${orgID}`
+    ).catch((e) => {
+      printError(e.message, "Error fetching contexts", false, 2);
+      printError("Skipping contexts", "Warning", true, 2);
+      return [];
+    });
+    printMessage(`${contexts.length}`, "Contexts found:");
+
+    // Potential here for too many requests
```

Comment:
I agree. This should definitely be a `for...of` instead of a `map`.  The former will ensure the request will occur in sequence, reducing the change of a 429. `map` with `Promise.all` will run them in parallel.

Reply:
I was doing this earlier but running into an issue where the async calls within were not being waited for. Ill take another look here.

Reply:
fixed